### ISSUE TITLE
onFocusChange setting is not respected

### ISSUE THREAD

**Issue by anavarre** [eclipse-theia/theia#11364]
### Bug Description:

When configurating Theia to automatically save `onFocusChange`, it is expected that the **Auto Save Delay** setting will not have any effect since it is written very clearly that this setting gets applied only when Auto Save is set to `afterDelay`.

<img width="662" alt="onFocusChange" src="https://user-images.githubusercontent.com/293478/176417188-ad2f10a6-6408-4246-8f06-26b9f635f306.png">

However, it seems this is not respected as you can see in the GIF file below.

![bug-observation](https://user-images.githubusercontent.com/293478/176419202-7bca4b13-2f3a-43f9-990a-fa6f05f65da8.gif)

### Steps to Reproduce:

1. Configure **Auto Save** to be `OnFocusChange`
2. Set a high value for **Auto Save Delay**, e.g. `10000` to observe your file is never changed before 10 seconds, even when you change focus and expect the file to be saved then.
3. Continue playing with the **Auto Save Delay** value until you get convinced it is indeed not respecting the promise that `onFocusChange` should take precedence and **Auto Save Delay** should only be enforced when the **Auto Save** is set to `afterDelay`.

### Additional Information

- Operating System: Ubuntu Xenial (with ESM)
- Theia Version: 1.26

**Comment by colin-grant-work** [comment]
I believe that this is a duplicate of #10812.

**Comment by anavarre** [comment]
I did stumble upon #10812 before filing this issue but it didn't seem related at first sight.

**Comment by colin-grant-work** [comment]
> I did stumble upon #10812 before filing this issue but it didn't seem specifically related to it at first sight.

Yes, my fault - it's an issue more for internal consumption than public clarity. I'll update the headline to be clearer.
